### PR TITLE
Features

### PR DESCRIPTION
**Pull request by philn123**
I added a basic controls menu at the start of the game. I have a quest npc that has two quests and they are tracked when you accept them.

### CODE REVIEW

File: Assets/Scripts/Controllers/EventManager.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class EventManager : MonoBehaviour
+{
+    //[SerializeField]
+    //private QuestGiver questGiver = null;
```
Comment (randymester) [inline Assets/Scripts/Controllers/EventManager.cs]: Remove unneeded comments

File: Assets/Scripts/Controllers/EventManager.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+    [SerializeField]
+    private PlayerJournal playerJournal = null;
+
+    public void OnEnemyDeath() =>
+        playerJournal.OnActionCompleted(QuestType.Combat);
+
+
```
Comment (randymester) [inline Assets/Scripts/Controllers/EventManager.cs]: Clean up spacing

File: Assets/Scripts/Controllers/PlayerController.cs [hunk 3]
```diff
@@ -26,8 +26,6 @@ void Update()
         }
```
Comment (randymester) [inline Assets/Scripts/Controllers/PlayerController.cs]: Clean up spacing

File: Assets/Scripts/Controllers/PlayerController.cs [hunk 4]
```diff
@@ -26,8 +26,6 @@ void Update()
...
 
 
-
-
         if(Input.GetMouseButtonDown(1)){
```
Comment (randymester) [inline Assets/Scripts/Controllers/PlayerController.cs:29]: Clean up spacing

File: Assets/Scripts/Interactable.cs [hunk 5]
```diff
@@ -12,7 +12,7 @@ public class Interactable : MonoBehaviour
 
     public virtual void Interact(){
         //method is going to be overwritten\
-        //Debug.Log("Interacting with " + transform.name);
+        Debug.Log("Interacting with " + transform.name);
     }
```
Comment (randymester) [inline Assets/Scripts/Interactable.cs]: Remove debugging elements in final product. You could alternatively use the unity preprocessor directive "if UNITY_EDITOR"

File: Assets/Scripts/Inventory/Inventory.cs [hunk 6]
```diff
@@ -21,6 +21,9 @@ public class Inventory : MonoBehaviour
     public delegate void OnItemChanged();
     public OnItemChanged onItemChangedCallback;
 
+    [SerializeField]
+    private EventManager eventManager = null;
+
```
Comment (randymester) [inline Assets/Scripts/Inventory/Inventory.cs:27]: Clean up spacing

File: Assets/Scripts/New Features/ControlsUI.cs [hunk 7]
```diff
@@ -0,0 +1,24 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class ControlsUI : MonoBehaviour
+{
+
```
Comment (randymester) [inline Assets/Scripts/New Features/ControlsUI.cs:7]: Spacing

File: Assets/Scripts/New Features/ControlsUI.cs [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    public GameObject controlsUI;
+    // Start is called before the first frame update
+    //void Start()
+    //{
+        
+    //}
+
+    // Update is called once per frame
```
Comment (randymester) [inline Assets/Scripts/New Features/ControlsUI.cs:15]: Remove unneeded comments and boilerplate comments

File: Assets/Scripts/New Features/ControlsUI.cs [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+    void Update()
+    {
+        if (Input.GetButtonDown("Controls"))
+        {
+            controlsUI.SetActive(!controlsUI.activeSelf);
+        }
+
```
Comment (randymester) [inline Assets/Scripts/New Features/ControlsUI.cs:22]: Spacing

File: Assets/Scripts/New Features/PlayerJournal.cs [hunk 10]
```diff
@@ -0,0 +1,93 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerJournal : MonoBehaviour
+{
+
```
Comment (randymester) [inline Assets/Scripts/New Features/PlayerJournal.cs:7]: Spacing

File: Assets/Scripts/New Features/PlayerJournal.cs [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+    private const string progressFormat = "{0}/{1}";
+
+    [SerializeField]
+    private GameObject playerJournal = null;
+    // Start is called before the first frame update
```
Comment (randymester) [inline Assets/Scripts/New Features/PlayerJournal.cs:12]: Remove comment

File: Assets/Scripts/New Features/PlayerJournal.cs [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+    public List<Quest> PlayerQuests = null;
+
+    [SerializeField]
+    private GameObject panel = null;
+
+    [SerializeField]
+    private GameObject questItemPrefab = null;
+
+    #region Singleton
+    public static PlayerJournal instance;
```
Comment (randymester) [inline Assets/Scripts/New Features/PlayerJournal.cs:23]: Initialize this

File: Assets/Scripts/New Features/PlayerJournal.cs [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+    private void Awake()
+    {
+        if (instance != null)
+        {
+            Debug.LogWarning("More than one instance");
+            return;
+        }
```
Comment (randymester) [inline Assets/Scripts/New Features/PlayerJournal.cs:30]: -Debug logs should only be present in Unity Editor, see previous comment
-Also, if there is more than one instance, don't we want to destroy the new one if this is a singleton?

File: Assets/Scripts/New Features/PlayerJournal.cs [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+
+        controllers = new Dictionary<QuestType, QuestItemController>()
+        {
+            { QuestType.Combat, null },
+            { QuestType.Fetch, null },
+        };
+    }
+    #endregion
+
+    private Dictionary<QuestType, QuestItemController> controllers;
```
Comment (randymester) [inline Assets/Scripts/New Features/PlayerJournal.cs:44]: Move this field declaration up with the other fields. Also initialize it

File: Assets/Scripts/New Features/PlayerJournal.cs [hunk 15]
```diff
@@ -0,0 +1,93 @@
...
+
+
+
+    // Update is called once per frame
```
Comment (randymester) [inline Assets/Scripts/New Features/PlayerJournal.cs:48]: Spacing and delete boilerplate comments

File: Assets/Scripts/New Features/PlayerJournal.cs [hunk 16]
```diff
@@ -0,0 +1,93 @@
...
+    void Update()
+    {
+        if(Input.GetButtonDown("Journal"))
+        {
+            playerJournal.SetActive(!playerJournal.activeSelf);
+        }
+
```
Comment (randymester) [inline Assets/Scripts/New Features/PlayerJournal.cs:55]: Spacing

File: Assets/Scripts/New Features/PlayerJournal.cs [hunk 17]
```diff
@@ -0,0 +1,93 @@
...
+        {
+            playerJournal.SetActive(!playerJournal.activeSelf);
+        }
+        
+    }
+
+    public void Add(Quest quest)
+    {
+        PlayerQuests.Add(quest);
+        //Debug.Log(PlayerQuests[0].Title);
```
Comment (randymester) [inline Assets/Scripts/New Features/PlayerJournal.cs:61]: Remove commented out code or add a preprocessor directive around this

File: Assets/Scripts/New Features/PlayerJournal.cs [hunk 18]
```diff
@@ -0,0 +1,93 @@
...
+        GameObject newQuest = Instantiate(questItemPrefab) as GameObject;
+        QuestItemController controller = newQuest.GetComponent<QuestItemController>();
+        controllers[quest.QuestType] = newQuest.GetComponent<QuestItemController>(); ;
+        controller.Title.text = quest.Title;
+        controller.Description.text = quest.Description;
+        controller.ProgressText.text = quest.ProgressText;
+        newQuest.transform.parent = panel.transform;
+        newQuest.transform.localScale = Vector3.one;
+    }
+    public void OnActionCompleted(QuestType questType)
```
Comment (randymester) [inline Assets/Scripts/New Features/PlayerJournal.cs:71]: Add a space between these

File: Assets/Scripts/New Features/PlayerJournal.cs [hunk 19]
```diff
@@ -0,0 +1,93 @@
...
+                    progressFormat,
+                    quest.ActionsCompleted,
+                    quest.ActionsNeeded));
+
+            }
+
+        }
+
+    }
+
```
Comment (randymester) [inline Assets/Scripts/New Features/PlayerJournal.cs:92]: Spacing

File: Assets/Scripts/New Features/PlayerJournal.cs [hunk 20]
```diff
@@ -0,0 +1,93 @@
...
+        newQuest.transform.localScale = Vector3.one;
+    }
+    public void OnActionCompleted(QuestType questType)
+    {
+        foreach(Quest quest in PlayerQuests)
+        {
+            if(quest.QuestType == questType && quest.OnActionCompleted())
+            {
+                Debug.Log("Action Completed");
+            }
```
Comment (randymester) [inline Assets/Scripts/New Features/PlayerJournal.cs:78]: Refer to previous debug log comments

File: Assets/Scripts/New Features/QuestGiver.cs [hunk 21]
```diff
@@ -0,0 +1,114 @@
...
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.Assertions;
+using TMPro;
+using System.Linq;
+
+public class QuestGiver : Interactable
+{
+    private const string progressFormat = "{0}/{1}";
+
```
Comment (randymester) [inline Assets/Scripts/New Features/QuestGiver.cs:12]: Remove unused fields

File: Assets/Scripts/New Features/QuestGiver.cs [hunk 22]
```diff
@@ -0,0 +1,114 @@
...
+    [SerializeField]
+    private Canvas canvas = null;
+    //each quest giver needs a list of quests
+    //need to open a panel that loads those quests
```
Comment (randymester) [inline Assets/Scripts/New Features/QuestGiver.cs:16]: What is this a comment for? Put comments directly above what they are referring to

File: Assets/Scripts/New Features/QuestGiver.cs [hunk 23]
```diff
@@ -0,0 +1,114 @@
...
+                actionsNeeded: 1));
+
+        quests.Add(new Quest(
+          title: "Pick up Equipment",
+          description: "Pick up Helmet, Plate Armor, Plate Leg, Sword, and Shield",
+          progressTest: "0/5",
+          questType: QuestType.Fetch,
+          actionsNeeded: 5));
+
+
```
Comment (randymester) [inline Assets/Scripts/New Features/QuestGiver.cs:67]: Spacing

File: Assets/Scripts/New Features/QuestGiver.cs [hunk 24]
```diff
@@ -0,0 +1,114 @@
...
+        acceptButton.onClick.AddListener(() => OnAccept());
+        switchButton.onClick.AddListener(() => OnSwitch());
+
+        progressText.SetText("");
+
```
Comment (randymester) [inline Assets/Scripts/New Features/QuestGiver.cs]: Spacing

File: Assets/Scripts/New Features/QuestGiver.cs [hunk 25]
```diff
@@ -0,0 +1,114 @@
...
+        PlayerJournal.instance.Add(currentQuest);
+
+        //progressText.SetText(string.Format(
+        //    progressFormat,
+        //    currentQuest.ActionsCompleted,
+        //    currentQuest.ActionsNeeded));
+
+        //acceptButton.gameObject.SetActive(false);
+        //switchButton.gameObject.SetActive(false);
+
```
Comment (randymester) [inline Assets/Scripts/New Features/QuestGiver.cs]: Remove commented out code

File: Assets/Scripts/New Features/QuestGiver.cs [hunk 26]
```diff
@@ -0,0 +1,114 @@
...
+        if(quests.Count == 0)
+        {
+            canvas.enabled = false;
+        }
+        else
+        {
+            OnSwitch();
+        }
+
```
Comment (randymester) [inline Assets/Scripts/New Features/QuestGiver.cs]: Spacing

File: Assets/Scripts/New Features/QuestGiver.cs [hunk 27]
```diff
@@ -0,0 +1,114 @@
...
+    }
+
+    private void OnSwitch()
+    {
+        indexCurrentQuest = (indexCurrentQuest + 1) % quests.Count();
+        currentQuest = quests[indexCurrentQuest];
+        titleText.SetText(quests[indexCurrentQuest].Title);
+        descriptionText.SetText(quests[indexCurrentQuest].Description);
+
```
Comment (randymester) [inline Assets/Scripts/New Features/QuestGiver.cs]: Spacing

File: Assets/Scripts/New Features/QuestItemController.cs [hunk 28]
```diff
@@ -0,0 +1,11 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using TMPro;
+
+public class QuestItemController : MonoBehaviour
+{
+    public TextMeshProUGUI Title;
+    public TextMeshProUGUI Description;
+    public TextMeshProUGUI ProgressText;
```
Comment (randymester) [inline Assets/Scripts/New Features/QuestItemController.cs]: Initialize all fields

**Review by randymester — CHANGES_REQUESTED** [review]
Your code and logic look good, you just have to clean it up

**Comment by philn123** [comment]
Hey Randy, I pushed the changes that you requested

**Review by randymester — APPROVED** [review]

**Pull request merged**